src/utils/DataLoader.js

```diff
@@ -41,26 +41,94 @@ async function load(metadata) {
 
     points.reverse();
     return points;
-    // return new Promise((resolve) => {
-    //     setTimeout(function () {
-    //         resolve();
-    //     }, 10000);
-    // }).then(function () {
-    //     return points;
-    // });
+}
+
+// Merges two sorted arrays into one. Key is timestamp attribute
+function merge(A, B){
+    let i = 0;
+    let j = 0;
+    let C = [];
+    while(i < A.length && j < B.length) {
+        if(new Date(A[i]['timestamp']) < new Date(B[j]['timestamp'])) {
+            C.push(A[i]);
+            i++;
+        } else {
```

Comment:
Instead of nesting if inside else:
`
else {
 if {} 
 else {} 
}
`
I suggest using:
`
else if {}
else {}
`